src/main/kotlin/persistence/data/Door43Retrieval.kt

```diff
@@ -0,0 +1,46 @@
+package persistence.data
+
+import data.Language
+
+class Door43Retrieval(val api: Api = Api()) {
+
+    fun getLanguages() : List<persistence.data.Lang>{
+        val execute = api.getLang("").execute()
+        return execute.body()
+    }
+
+    fun getNames() : Array<String>{
+        val execute = api.getLang("").execute()
+        var i = 0
+        var size = execute.body().size
+        var names = Array(size) { "n = $it" }
+        //var names : Array<String> = arrayOf<String>()
+        for(langName in execute.body()){
+            names[i] = langName.ln
+        }
+        return names
+    }
+
+    fun getSlugs() : Array<String>{
+        val execute = api.getLang("").execute()
+        var i = 0
+        var size = execute.body().size
+        var slugs = Array(size) { "n = $it" }
```

Comment:
if you absolutely need to use an Array, you can to .toArray from a list; this allows you to have a dynamically resizing list that you can just add to